core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,444 @@
+        KiwixDialog.NotesDiscardConfirmation,
+        ::dismissAddNoteDialog
+      )
+    } else {
+      // Closing unedited note dialog straightaway
+      dismissAddNoteDialog()
+    }
+  }
+
+  private fun disableMenuItems() {
+    if (toolbar.menu != null) {
+      saveItem.isEnabled = false
+      shareItem.isEnabled = false
+      deleteItem.isEnabled = false
+      saveItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+      shareItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+      deleteItem.icon.alpha = DISABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableSaveNoteMenuItem() {
+    if (toolbar.menu != null) {
+      saveItem.isEnabled = true
+      saveItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableDeleteNoteMenuItem() {
+    if (toolbar.menu != null) {
+      deleteItem.isEnabled = true
+      deleteItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  private fun enableShareNoteMenuItem() {
+    if (toolbar.menu != null) {
+      shareItem.isEnabled = true
+      shareItem.icon.alpha = ENABLE_ICON_ITEM_ALPHA
+    } else {
+      Log.d(TAG, "Toolbar without inflated menu")
+    }
+  }
+
+  override fun onViewCreated(
+    view: View,
+    savedInstanceState: Bundle?
+  ) {
+    super.onViewCreated(view, savedInstanceState)
+    toolbar.setTitle(R.string.note)
+    toolbar.setNavigationIcon(R.drawable.ic_close_white_24dp)
+    toolbar.setNavigationOnClickListener {
+      exitAddNoteDialog()
+      closeKeyboard()
+    }
+    toolbar.setOnMenuItemClickListener { item: MenuItem ->
+      when (item.itemId) {
+        R.id.share_note -> shareNote()
+        R.id.save_note -> saveNote(add_note_edit_text.text.toString())
+        R.id.delete_note -> deleteNote()
+      }
+      true
+    }
+    toolbar.inflateMenu(R.menu.menu_add_note_dialog)
+    // 'Share' disabled for empty notes, 'Save' disabled for unedited notes
+    disableMenuItems()
+    add_note_text_view.text = articleTitle
+
+    // Show the previously saved note if it exists
+    displayNote()
+    add_note_edit_text.addTextChangedListener(object : TextWatcher {
+      @SuppressWarnings("EmptyFunctionBlock")
+      override fun beforeTextChanged(
+        s: CharSequence,
+        start: Int,
+        count: Int,
+        after: Int
+      ) {
+      }
+
+      override fun onTextChanged(
+        s: CharSequence,
+        start: Int,
+        before: Int,
+        count: Int
+      ) {
+        noteEdited = true
+        enableSaveNoteMenuItem()
+        enableShareNoteMenuItem()
+      }
+
+      @SuppressWarnings("EmptyFunctionBlock")
+      override fun afterTextChanged(s: Editable) {
+      }
+    })
+    if (!noteFileExists) {
+      // Prepare for input in case of empty/new note
+      add_note_edit_text.requestFocus()
+      showKeyboard()
```

Comment:
I think we have some extension functions for keyboard management

Reply:
We do not, unless they are coming from androidx-ktx but I don't think that exists